app/src/main/java/com/example/movieapp/MovieViewHolder.kt

```diff
@@ -0,0 +1,77 @@
+package com.example.movieapp
+
+import android.util.Log
+import android.view.View
+import androidx.recyclerview.widget.RecyclerView
+import com.example.movieapp.crawler.pojo.Movie
+import com.example.movieapp.databinding.ListItemBinding
+import com.example.movieapp.likesmanager.App
+import com.squareup.picasso.Picasso
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.MainScope
+import kotlinx.coroutines.launch
+
+/**
+ * Created by Benjamin Vouillon on 15,July,2020
+ */
+const val PURL = "https://image.tmdb.org/t/p/w185"
+private val TAG = "MovieViewHolder"
+
+class MovieViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
+    private val binding = ListItemBinding.bind(itemView)
+    private var movie: Movie? = null
+
+
+    fun bind(movie: Movie) {
+
+        this.movie = movie
+
+        binding.apply {
+            itemView.resources.apply {
+
+                listTitle.text = movie.title
+
+                listPopularity.text =
+                    getString(R.string.itemPopularity, movie.popularity.toString())
+
+                listRate.text =
+                    getString(R.string.itemRate, movie.voteAverage.toString())
+
+                Picasso.get().load(PURL + movie.posterPath)
+                    .into(image)
+            }
+        }
+        binding.likeBtn.setOnClickListener(LikeClicker(movie, binding))
+    }
+
+    class LikeClicker(val movie: Movie, private val binding: ListItemBinding) :
+        View.OnClickListener {
+
+        init {
+            MainScope().launch(Dispatchers.IO) {
+                if (App.database.movieDAO().isLiked(movie.id)) {
```

Comment:
It is not the ViewHolder job to do this kind of job -> move it to you viewModel

Reply:
The Database Call ?

Reply:
Or just the init ?

Reply:
Done